### PR TITLE
Refactor IndexFieldTypeProfileService to no longer extend PaginatedDbService

### PR DESCRIPTION
**Pull request by thll**
/nocl

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/indexer/indexset/profile/IndexFieldTypeProfileService.java [hunk 1]
```diff
@@ -77,31 +77,31 @@ public class IndexFieldTypeProfileService extends PaginatedDbService<IndexFieldT
...
-        this.db.createIndex(new BasicDBObject(IndexFieldTypeProfile.NAME_FIELD_NAME, 1), new BasicDBObject("unique", false));
-        this.profileCollection = mongoCollections.collection(INDEX_FIELD_TYPE_PROFILE_MONGO_COLLECTION_NAME, IndexFieldTypeProfile.class);
         this.indexSetService = indexSetService;
         this.dbQueryCreator = new DbQueryCreator(IndexFieldTypeProfile.NAME_FIELD_NAME, ATTRIBUTES);
         this.indexFieldTypeProfileUsagesService = indexFieldTypeProfileUsagesService;
+
+        collection = mongoCollections.collection(INDEX_FIELD_TYPE_PROFILE_MONGO_COLLECTION_NAME, IndexFieldTypeProfile.class);
+        mongoUtils = mongoCollections.utils(collection);
+
+        collection.createIndex(Indexes.ascending(IndexFieldTypeProfile.NAME_FIELD_NAME));
```
Comment (luk-kaminski) [inline graylog2-server/src/main/java/org/graylog2/indexer/indexset/profile/IndexFieldTypeProfileService.java:97]: Originally index was unique. It seems that you have forgotten about it when refactoring.

Comment (thll) [reply graylog2-server/src/main/java/org/graylog2/indexer/indexset/profile/IndexFieldTypeProfileService.java:97]: 🤔 was it? It looks like it was explicitly set to not unique, which I think is the default. Or am I missing something?

Comment (luk-kaminski) [reply graylog2-server/src/main/java/org/graylog2/indexer/indexset/profile/IndexFieldTypeProfileService.java:97]: Oh, sorry, my mistake...

**Review by luk-kaminski — APPROVED** [review]

**Comment by thll** [comment]
thank you for reviewing, @luk-kaminski!

**Pull request merged**